Adds xAmzHeadersAtCancel parameter to Evaporate#add

bikeath1337 opened the pull request:
This PR rebases work by @jakubzitny and adds test coverage. Supersedes #232.

@jakubzitny please feel free to adapt the specs. I'm not really the up to speed on more elegant ways to reuse and write the expects.

jakubzitny commented:
@bikeath1337 thanks for the implementation.

One more thing should definitely be part of this MR and that is headers for `listUploadParts` and `getUploadParts` requests. It was not part of the original MR but it could look something like this https://github.com/jakubzitny/EvaporateJS/commit/f917d6b39a04e0816527861b7c77b51f89e24af7.

bikeath1337 commented:
@jakubzitny I reworked this to unify the headers into a single parameter. I think this is correct. The new option `xAmzHeadersCommon` apply to ALL existing requests and will override the use of xAmzHeadersAtUpload and xAmzHeadersAtComplete (These two parameters presumably are deprecated at some point in the future). xAmzHeadersAtInitiate remain as-is as the initiate request has special needs.

I added skipped specs for the paths that use localStorage as I don't know how to spec these out yet.

bikeath1337 commented on test/request_headers.spec.js:
```diff
@@ -0,0 +1,153 @@
+}
+
+let server,
+    requestHeaders = {}
```
@jakubzitny I need to keep `requestHeaders` local to each test. What's the best way to do this if these tests run asynchronously?

jakubzitny replied:
I'd keep the declaration global and initialize it in `test.beforeEach()` which is run before each test.

```js
test.before(() => {
  requestHeaders = {}
})
```

jakubzitny replied:
This looks great, do you have any other problems with tests?

bikeath1337 replied:
Thanks. I updated the specs just now.

I don't have any other pressing issues other than the need to test more and the biggest issue is to reduce the test code bloat, IMO...in the future.
